README.md

```diff
@@ -66,6 +66,17 @@ If you want to use MQTT on WebSocket, you need to define `MQTT_USE_WS` macro. mq
     * WebSocket
       * [example/tls_ws_both.cpp](https://github.com/redboltz/mqtt_cpp/blob/master/example/tls_ws_both.cpp)
 
+## Usage in cmake project
+
+Add following lines to your `CMakeList.txt`
+
+```
+find_package(mqtt_cpp REQUIRED)
+target_link_libraries(${LIBNAME} LINK_PUBLIC mqtt_cpp::mqtt_cpp)
```

Comment:
What does `${LIBNAME}` mean?
If if is a users target name, it should be something like `${YOUR_TARGET}`.

Reply:
I've tested the PR on my environment.
I did `make install` mqtt_cpp. Then create the new cmake project.
The CMakeLists.txt is

```cmake
PROJECT (mytest)

find_package(mqtt_cpp REQUIRED)
target_link_libraries(${PROJECT} LINK_PUBLIC mqtt_cpp::mqtt_cpp)
```

I did

```sh
make build
cd build
cmake -DCMAKE_FIND_DEBUG_MODE=1 ..
```

Then I got the following error:

```
CMake Error at CMakeLists.txt:4 (target_link_libraries):
  Cannot specify link libraries for target "LINK_PUBLIC" which is not built
  by this project.
```

Reply:
Thanks for the input. I improved the readme. Apparently the target is missing that you add the `target_link_libraries` to. That means a target generation command like
```
add_library(${PROJECT_NAME} <dependencies>)
```
or
```
add_executable(${PROJECT_NAME} <dependencies>)
```
is missing.

Reply:
The `target_link_libraries` call would then also need the following call

```
target_link_libraries(${PROJECT_NAME} mqtt_cpp::mqtt_cpp)
```
since the `PROJECT` variable itself is not set.

Reply:
I still got the error. 

I don't understand the following code at readme.

```cmake
set(LIBNAME "your_project_name")
add_library(${LIBNAME} <your dependencies here>)
find_package(mqtt_cpp REQUIRED)
target_link_libraries(${LIBNAME} LINK_PUBLIC mqtt_cpp::mqtt_cpp)
```

The user creates library? I think user creates application.

Reply:
Could you give me complete and minimal example of `CMakeLists.txt` that uses installed `mqtt_cpp`?
I never success cmake, so far.

Reply:
Create a library that uses mqtt_cpp

```
PROJECT(my_mqtt)

MESSAGE(STATUS "Configuring ${PROJECT_NAME}")

FILE(GLOB SRC "*.cpp")

ADD_LIBRARY(my_mqtt ${SRC})
ADD_LIBRARY(my::mqtt ALIAS my_mqtt)

TARGET_INCLUDE_DIRECTORIES(my_mqtt PUBLIC ${MY_FRAMEWORKS_ROOT}/inc/)

TARGET_LINK_LIBRARIES(my_mqtt my::core my::log mqtt_cpp_iface Boost::system OpenSSL::SSL nlohmann_json::nlohmann_json )
```

Create an application that uses the library.

```
PROJECT(my-application)

MESSAGE(STATUS "Configuring ${PROJECT_NAME}")

FILE(GLOB SRC "*.cpp")

ADD_EXECUTABLE(my-application ${SRC})

TARGET_LINK_LIBRARIES(my-application my::mqtt Boost::program_options)

INSTALL(TARGETS my-application RUNTIME DESTINATION /usr/bin)
```

Reply:
What this PR does is make it so someone (e.g., a distribution packager for a linux distribution) can install mqtt_cpp via "make install", and them someone else who wants to use mqtt_cpp, in their CMakeLists.txt, can use the CMake "find_package" command to locate mqtt_cpp on the filesystem and use the code in their own application.

I imagine that this change will also make it easier to satisfy this issue: https://github.com/redboltz/mqtt_cpp/issues/77 (but that's just speculation on my part. I know nothing of vcpkg)